File: client/src/app/modules/quotation/services/jo-notice.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { JoNotice } from '../model/JoNotice';

@Injectable({
  providedIn: 'root'
})
export class JoNoticeService extends AppRestService<JoNotice> {

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getJoNoticeForAffaire(affaireId: number) {
    return this.getList(new HttpParams().set("affaireId", affaireId), "jo-notice/affaire");
  }
}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/JoNoticeService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.Affaire;
import com.jss.osiris.modules.osiris.quotation.model.JoNotice;

public interface JoNoticeService {

    public void updateJoNotices();

    public List<JoNotice> getJoNoticeByAffaire(Affaire affaire);

}
